charts/consul/values.yaml

```diff
@@ -162,6 +162,17 @@ global:
       # and check the name of `metadata.name`.
       consulClientRole: ""
 
+      # The Vault role for the Consul client snapshot agent.
+      # The role must be connected to the Consul client snapshot agent's service account and
+      # have a policy with read capabilities for the following secrets:
+      # - gossip encryption key defined by `client.snapshotAgent.configSecret.secretName`.
+      # To discover the service account name of the Consul client, run
+      #     ```shell-session
+      #     $ helm template --show-only templates/client-snapshot-agent- serviceaccount.yaml <release-name> charts/consul
+      #     ```
+      # and check the name of `metadata.name`.
+      consulSnapshotAgentRole: ""
```

Comment:
User has to set up a vault policy and auth role that maps the service account to the policy that has access to the snapshot agent config secret.  Configuring it here will assign it in a vault annotation in the snapshot agent deployment so that the vault injector for the snapshot agent config will be authorized.